# Portfolio assignment 8
- Calculate the 90%, 95%, 99% and 99,99% confidence interval for at least 2 columns with numerical data in the dataset you already used. Do you see the impact the confidence has on the interval?
- Now calculate the 95% confidence interval again but use only the first 10% of your rows. Compare this interval to the previous 95% confidence interval you calculated. Do you see the impact of having less data?

In [5]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import scipy.stats as st

In [6]:
df = pd.read_csv('Datasets/aug_test.csv', sep=',')
df.head()

enrollee_id      city city_development_index  gender  \
0        32403   city_41  8.270.000.000.000.000    Male   
1         9858  city_103                   0.92  Female   
2        31806   city_21                    624    Male   
3        27385   city_13  8.270.000.000.000.000    Male   
4        27724  city_103                   0.92    Male   

       relevent_experience enrolled_university education_level  \
0  Has relevent experience    Full time course        Graduate   
1  Has relevent experience       no_enrollment        Graduate   
2   No relevent experience       no_enrollment     High School   
3  Has relevent experience       no_enrollment         Masters   
4  Has relevent experience       no_enrollment        Graduate   

  major_discipline experience company_size company_type last_new_job  \
0             STEM          9          <10          NaN            1   
1             STEM          5          NaN      Pvt Ltd            1   
2              NaN         <1          NaN      Pvt Ltd        never   
3             STEM         11        10/49      Pvt Ltd            1   
4             STEM        >20       10000+      Pvt Ltd           >4   

   training_hours  
0              21  
1              98  
2              15  
3              39  
4              72

## Confidence 90%

In [8]:
confidence = 0.90
st.t.interval(confidence, len(df)-1, loc=df['training_hours'].mean(), scale=st.sem(df['training_hours']))

(62.834746612653554, 67.1314346931238)

## Confidence 95%

In [9]:
confidence = 0.95
st.t.interval(confidence, len(df)-1, loc=df['training_hours'].mean(), scale=st.sem(df['training_hours']))

(62.42283904308297, 67.54334226269438)

## Confidence 99%

In [10]:
confidence = 0.99
st.t.interval(confidence, len(df)-1, loc=df['training_hours'].mean(), scale=st.sem(df['training_hours']))

(61.617244602159374, 68.34893670361798)

## Confidence 99,99%

In [11]:
confidence = 0.9999
st.t.interval(confidence, len(df)-1, loc=df['training_hours'].mean(), scale=st.sem(df['training_hours']))

(59.8941536510212, 70.07202765475616)

## Impact Cofidence
The interval becomes large everytime the confidence gets higher.

## Confidence 95% with first 10% rows

In [13]:
print(len(df.index))

2129


There are 2129 rows in this dataset, 10% of the rows is 213

In [14]:
df1 = df.head(213)
confidence = 0.95
st.t.interval(confidence, len(df1)-1, loc=df1['training_hours'].mean(), scale=st.sem(df1['training_hours']))

(54.08622552271199, 69.74476039278098)

## Impact less data
The interval by less data becomes bigger.